src/she/sdl2/sdl2_surface.cpp

```diff
@@ -0,0 +1,431 @@
+// SHE library
+// Copyright (C) 2012-2016  David Capello
+//
+// This file is released under the terms of the MIT license.
+// Read LICENSE.txt for more information.
+
+#include <SDL2/SDL_blendmode.h>
+#include <SDL2/SDL_rect.h>
+#include <SDL2/SDL_surface.h>
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
+#include "she/sdl2/sdl2_surface.h"
+
+#include "base/string.h"
+#include "gfx/point.h"
+#include "gfx/rect.h"
+
+#include <iostream>
+#include <SDL2/SDL.h>
+
+namespace {
+
+  void checked_mode(int offset)
+  {
+    // static BITMAP* pattern = NULL;
+    // int x, y, fg, bg;
+
+    // if (offset < 0) {
+    //   if (pattern) {
+    //     destroy_bitmap(pattern);
+    //     pattern = NULL;
+    //   }
+    //   drawing_mode(DRAW_MODE_SOLID, NULL, 0, 0);
+    //   return;
+    // }
+
+    // if (!pattern)
+    //   pattern = create_bitmap(8, 8);
+
+    // bg = makecol(0, 0, 0);
+    // fg = makecol(255, 255, 255);
+    // offset = 7 - (offset & 7);
+
+    // clear_bitmap(pattern);
+
+    // for (y=0; y<8; y++)
+    //   for (x=0; x<8; x++)
+    //     putpixel(pattern, x, y, ((x+y+offset)&7) < 4 ? fg: bg);
+
+    // drawing_mode(DRAW_MODE_COPY_PATTERN, pattern, 0, 0);
+  }
+
+}
+
+namespace she {
+
+  namespace sdl {
+    extern SDL_Surface* screen;
+  }
+
+  inline int to_sdl(SDL_PixelFormat* format, gfx::Color color)
+  {
+    return SDL_MapRGBA(format, gfx::getr(color), gfx::getg(color), gfx::getb(color), gfx::geta(color));
+  }
+
+  inline gfx::Color from_sdl(SDL_PixelFormat *format, int color)
+  {
+    return gfx::rgba(
+      (color & format->Rmask) >> format->Rshift << format->Rloss,
+      (color & format->Gmask) >> format->Gshift << format->Gloss,
+      (color & format->Bmask) >> format->Bshift << format->Bloss,
+      (color & format->Amask) >> format->Ashift << format->Aloss
+      );
+  }
```

Comment:
nitpicking, but I expect something called "from" coming before something called "to"